appcues/src/main/java/com/appcues/trait/appcues/TargetInteractionTrait.kt

```diff
@@ -0,0 +1,91 @@
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction.InteractionType.TARGET_TAPPED
+import com.appcues.data.model.AppcuesConfigMap
+import com.appcues.data.model.getConfigActions
+import com.appcues.trait.BackdropDecoratingTrait
+import com.appcues.trait.extensions.getRect
+import com.appcues.trait.extensions.rememberTargetRectangleInfo
+import com.appcues.ui.composables.AppcuesActionsDelegate
+import com.appcues.ui.composables.LocalAppcuesStepMetadata
+import com.appcues.ui.extensions.toLongPressMotionOrNull
+import com.appcues.ui.extensions.toTapMotionOrEmpty
+import com.appcues.ui.utils.rememberAppcuesWindowInfo
+
+internal class TargetInteractionTrait(
+    override val config: AppcuesConfigMap,
+    private val actionProcessor: ActionProcessor,
+    actionRegistry: ActionRegistry,
+) : BackdropDecoratingTrait {
+
+    companion object {
+
+        const val TYPE = "@appcues/target-interaction"
+
+        private const val VIEW_DESCRIPTION = "Target Rectangle"
+    }
+
+    private val actions = config.getConfigActions("actions", actionRegistry)
+
+    private val actionDelegate = object : AppcuesActionsDelegate {
+        override fun onActions(actions: List<ExperienceAction>, interactionType: InteractionType, viewDescription: String?) {
+            actionProcessor.process(actions, interactionType, viewDescription)
+        }
+    }
+
+    @OptIn(ExperimentalFoundationApi::class)
+    @Composable
+    override fun BoxScope.BackdropDecorate(content: @Composable BoxScope.() -> Unit) {
+        val targetRectInfo = rememberTargetRectangleInfo(LocalAppcuesStepMetadata.current)
+
+        val rect = targetRectInfo.getRect(rememberAppcuesWindowInfo()) ?: Rect(Offset(0f, 0f), Size(0f, 0f))
+        val sizeDp = DpSize(rect.width.dp, rect.height.dp)
+        val offsetDp = DpOffset(rect.left.dp, rect.top.dp)
+
+        // calling content before our composable makes so we are on top of all other backdrop traits
+        // wrapping content call
+        content()
+
+        Box(
+            modifier = Modifier
+                .size(sizeDp)
+                .offset(x = offsetDp.x, y = offsetDp.y)
+                // add click listener but without any ripple effect.
+                .then(
+                    if (actions.isNotEmpty()) {
+                        Modifier.combinedClickable(
+                            interactionSource = remember { MutableInteractionSource() },
+                            indication = null,
+                            onLongClick = actions.toLongPressMotionOrNull(actionDelegate, TARGET_LONG_PRESSED, VIEW_DESCRIPTION),
+                            onClick = actions.toTapMotionOrEmpty(actionDelegate, TARGET_TAPPED, VIEW_DESCRIPTION),
+                        )
+                    }
+                    // when no actions are listed, the default behavior is to
+                    else Modifier.pointerInput(Unit) { detectTapGestures { } }
+                )
+        )
+    }
```

Comment:
can this composition be avoided entirely in the case of no target rect in metadata?

Reply:
thats a good catch, did some changes here that might cover this scenario